### PR TITLE
ZIO Test: Implement TestAspect#timed

### PR DESCRIPTION
**Pull request by adamgfraser**
Following up on discussions on Gitter, there was interest in having an easy way to see which tests were taking longer to run. This PR adds a new `TestAspect` timed that reports the execution time of each test as it runs. We could also potentially do something more sophisticated like gather all the execution times, rank order them, and show what percent of total time they took so you could easily see an 80/20 analysis of what is driving your total testing time, but that is not implemented here. I also added a new subtype of `TestAspect`, `perTestLabeled`, that modifies each test but also provide access to the test label.

### CODE REVIEW

File: test/shared/src/main/scala/zio/test/TestAspect.scala [hunk 1]
```diff
@@ -280,6 +281,25 @@ object TestAspect extends TimeoutVariants {
...
+    new TestAspect.PerTestLabeled[Nothing, Live[Clock] with Live[Console], Nothing, Any, Nothing, Any] {
+      def perTestLabeled[R <: Live[Clock] with Live[Console], E, S, L](
+        label: L,
+        test: ZIO[R, TestFailure[E], TestSuccess[S]]
+      ): ZIO[R, TestFailure[E], TestSuccess[S]] =
+        withLive(test.either)(_.timed).flatMap {
+          case (duration, ees) =>
+            live {
+              console.putStrLn {
+                s"$label took ${duration.toMillis / 1000.0} seconds"
```
Comment (ghostdogpr) [inline test/shared/src/main/scala/zio/test/TestAspect.scala]: I noticed we have similar code to render durations in `TimeoutVariants#renderDuration` and `DefaultTestReporter#logStats`. How about extracting it and using the same one everywhere? `TimeoutVariants` is probably the best one.

Comment (adamgfraser) [reply test/shared/src/main/scala/zio/test/TestAspect.scala]: Good idea. Will do that tomorrow.

**Comment by mschuwalow** [comment]
Very nice!
I've looked into this as well after our discussion on gitter and I feel maybe printing after every test is not the right way to do this. @iravid also requested the option to have resources that are only allocated once per suite and I think those two can be easily combined.
This would it also make it trivial to postprocess results like the ranking you suggested.
I don't think this can be expressed by the current formulation of TestAspect but maybe you have some idea.

**Comment by jdegoes** [comment]
I have been thinking about this. I have some code to help, basically a `TestAnnotation` so you can add arbitrary annotations (like timings) to each test, without altering any data structures. I'll try to get that PR up tomorrow and we can see if it might be useful for this.

**Comment by adamgfraser** [comment]
@jdegoes That sounds interesting! I wonder if we could use it to add information about the random seed to each test and then if it was a failing property based test the test reporter could report it.

**Comment by adamgfraser** [comment]
Closing in favor of #1951.

**Pull request closed without merging**